src/Wellcome.Dds/WorkflowProcessor/WorkflowProcessorService.cs

```diff
@@ -415,45 +441,65 @@ private async Task PollForWorkflowJobs(CancellationToken cancellationToken)
             logger.LogInformation("Cancellation requested in WorkflowProcessor, shutting down.");
         }
 
-        private async Task PollQueue(string queueUrl, DdsInstrumentationContext dbContext, CancellationToken cancellationToken)
+        private async Task PollQueues(Dictionary<string, string> queues, DdsInstrumentationContext dbContext, CancellationToken cancellationToken)
         {
-            if (queueUrl.IsNullOrWhiteSpace())
+            if (queues.Keys.Count == 0)
             {
+                logger.LogInformation("No queues configured to poll");
                 return;
             }
-            
-            var response = await sqsClient.ReceiveMessageAsync(new ReceiveMessageRequest
-            {
-                QueueUrl = queueUrl,
-                WaitTimeSeconds = 5,
-                MaxNumberOfMessages = 10,
-            }, cancellationToken);
-            var messageCount = response.Messages?.Count ?? 0;
-            if (messageCount > 0)
+
+            foreach (var queue in queues)

```

Comment:
Is the expectation here that there will be 1 processor listening to BD + Digitised queues? Or 2 different processors running, each listening to a different queue?

Reply:
Just one, that checks each queue in turn